azure-cognitiveservices-vision-computervision/README.md

```diff
@@ -0,0 +1,287 @@
+* Moderate images in content: An image can be analyzed for adult content.
+* Extract text from an image: An image can be scanned for text. In order to complete the analysis, Computer Vision may correct rotation before analysis. Computer Vision supports 25 languages, and automatically detects the language of extracted text.
+
+For more information about these resources, see [Working with Azure computer vision][computervision_resources].
+
+## Examples
+
+The following sections provide several code snippets covering some of the most common Computer Vision tasks, including:
+
+* [Analyze an image](#analyze-an-image)
+* [Analyze an image by domain](#analyze-an-image-by-domain)
+* [Get text description of an image](#get-text-description-of-an-image)
+* [Generate thumbnail](#generate-thumbnail)
+* [Get subject domain list](#get-subject-domain-list)
+
+### Analyze an image
+
+You can analyze an image for certain features. Use the `visual_features` property to set the types of analysis to perform on the image. Common values are `VisualFeatureTypes.tags` and `VisualFeatureTypes.description`.
+
+```Python
+url = "https://upload.wikimedia.org/wikipedia/commons/thumb/1/12/Broadway_and_Times_Square_by_night.jpg/450px-Broadway_and_Times_Square_by_night.jpg"
+
+image_analysis = client.analyze_image(url,visual_features=[VisualFeatureTypes.tags])
+
+for tag in image_analysis.tags:
+    print(tag)
+```
+
+
+### Analyze an image by domain
+
+You can analyze an image by subject domain. Get the [list of support subject domains](#get-subject-domain-list) in order to use the correct domain name.  
+
+```Python
+domain = "landmarks"
+url = "https://upload.wikimedia.org/wikipedia/commons/thumb/1/12/Broadway_and_Times_Square_by_night.jpg/450px-Broadway_and_Times_Square_by_night.jpg"
+language = "en"
+
+analysis = client.analyze_image_by_domain(domain, url, language)
+
+for landmark in analysis.result["landmarks"]:
+    print(landmark["name"])
+    print(landmark["confidence"])
+```
+
+### Get text description of an image
+
+You can get a language-based text description of an image. Request several descriptions with the `max_description` property if you are doing text analysis for keywords associated with the image. Examples of a text description for the following image include `a train crossing a bridge over a body of water`, `a large bridge over a body of water`, and `a train crossing a bridge over a large body of water`.
+
+```Python
+domain = "landmarks"
+url = "http://www.public-domain-photos.com/free-stock-photos-4/travel/san-francisco/golden-gate-bridge-in-san-francisco.jpg"
+language = "en"
+max_descriptions=3
+
+analysis = client.describe_image(url, max_descriptions, language)
+
+for caption in analysis.captions:
+    print(caption.text)
+    print(caption.confidence)
+```
+
+### Generate thumbnail
+
+You can generate a thumbnail (JPG) of an image. The thumbnail does not need to be in the same proportions as the original image. 
```

Comment:
Recommend adding a one-liner that states this example uses Pillow, with a link to its PyPi page. Something like:

This example uses the [Pillow](https://pypi.org/project/Pillow/) package to save the new thumbnail image locally.

Reply:
That is, I suppose, unless *everyone in the Python world* knows what PIL is and that you need Pillow. I hadn't used it before, and because the [PIL package](https://pypi.org/project/PIL/) itself seems to not have a straight `pip install` story, I had to poke around a bit to find out what/how to install to get the PIL import working.

Reply:
I'm going to leave it as pillow for now. If feedback from @annatisch is to change it, then I will. It could also be something that is asked in the user study. "If not pillow, what do you use?"